src/ert/config/field.py

```diff
@@ -325,6 +327,36 @@ def ny(self) -> int:
     def nz(self) -> int:
         return self.ertbox_params.nz
 
+    def calc_rho_for_2d_grid_layer(
+        self,
+        obs_xpos: npt.NDArray[np.double],
+        obs_ypos: npt.NDArray[np.double],
+        obs_main_range: npt.NDArray[np.double],
+        obs_perp_range: npt.NDArray[np.double],
+        obs_anisotropy_angle: npt.NDArray[np.double],
+        scaling_function: ScalingFunctions = ScalingFunctions.gaspari_cohn,
+    ) -> npt.NDArray[np.double]:
+        # Can only be used if ertbox coordinate system is defined
+        assert self.ertbox_params.nx > 0
```

Comment:
I think we can remove all the asserts here since this must be validated upfront before the experiment / update step starts running.